File: examples-sdk/csharp/028-basic-sse-hub-streaming/Main.cs
// 028-basic-sse-hub-streaming — C# SDK equivalent
// Compile: vil compile --from csharp --input 028-basic-sse-hub-streaming/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("sse-hub-demo", 8080);
var events = new ServiceProcess("events");
events.Endpoint("POST", "/publish", "publish");
events.Endpoint("GET", "/stream", "stream");
events.Endpoint("GET", "/stats", "stats");
server.Service(events);
server.Compile();

File: examples-sdk/java/028-basic-sse-hub-streaming/Main.java
// 028-basic-sse-hub-streaming — Java SDK equivalent
// Compile: vil compile --from java --input 028-basic-sse-hub-streaming/Main.java --release

import dev.vil.*;

public class Main {
    public static void main(String[] args) {
        VilServer server = new VilServer("sse-hub-demo", 8080);
        ServiceProcess events = new ServiceProcess("events");
        events.endpoint("POST", "/publish", "publish");
        events.endpoint("GET", "/stream", "stream");
        events.endpoint("GET", "/stats", "stats");
        server.service(events);
        server.compile();
    }
}
